Lib/logging/handlers.py

```diff
@@ -1015,8 +1015,22 @@ def emit(self, record):
                     self.socket.close()
                     self._connect_unixsocket(self.address)
                     self.socket.send(msg)
+            # UDP socket type 
             elif self.socktype == socket.SOCK_DGRAM:
                 self.socket.sendto(msg, self.address)
+            # TCP socket type
+            # This block doesn't exist in original handlers.py code for TCP 
+            elif self.socktype == socket.SOCK_STREAM:
+                try:
+
+                    self.socket.sendall(msg)
+                # create new TCP socket stream, if TCP socket broken or disconnected 
+                except OSError:
+
+                    self.socket = socket.socket(socket.AF_INET, self.socktype)
```

Comment:
While I just reformatted your issue, I'm not sure whether we should retry blindly. What happens here is that were are changing `self.socket` and the previous `self.socket` could not necessarily be freed. So we should probably do something like `self.socket.close()` and then `self.socket = socket.socket(...)`. Or use a private method for that as for the `unixsocket` case.